src/modules/dataman/dataman.cpp

```diff
@@ -1044,27 +1053,16 @@ _ram_flash_shutdown()
 static int
 _ram_flash_wait(px4_sem_t *sem)
 {
-	if (!dm_operations_data.ram_flash.flush_timeout_usec) {
+	if (!dm_operations_data.ram_flash.flush_timeout.tv_sec) {
 		px4_sem_wait(sem);
 		return 0;
 	}
 
-	const uint64_t now = hrt_absolute_time();
-
-	if (now >= dm_operations_data.ram_flash.flush_timeout_usec) {
-		_ram_flash_flush();
-		return 0;
-	}
-
-	const uint64_t diff = dm_operations_data.ram_flash.flush_timeout_usec - now;
-	struct timespec abstime;
-	abstime.tv_sec = diff / USEC_PER_SEC;
-	// FIXME: this could be made more performant.
-	abstime.tv_nsec = (diff % USEC_PER_SEC) * NSEC_PER_USEC;
+	int ret;
 
-	px4_sem_timedwait(sem, &abstime);
+	while ((ret = px4_sem_timedwait(sem, &dm_operations_data.ram_flash.flush_timeout)) == -1 && errno == EINTR);
```

Comment:
Doesn't `px4_sem_timedwait` on Posix need `CLOCK_MONOTONIC`?

https://github.com/PX4/Firmware/blob/07d656e971a72d1202651dfd3b4642736fb078d7/platforms/posix/src/px4/common/px4_sem.cpp#L65

Reply:
From what I see `sem_timedwait` needs `CLOCK_REALTIME` (not talking about pthread locks), so we would need to abstract that depending on what underlying `sem_t` is used.
In any case this is only used on NuttX (and uses NuttX-specific FLASH API calls), and does not make much sense on POSIX.

Reply:
Oh right!